library/Icinga/Web/Controller/ActionController.php

```diff
@@ -519,7 +519,19 @@ protected function postDispatchXhr()
         }
 
         if ($this->autorefreshInterval !== null) {
-            $resp->setAutoRefreshInterval($this->autorefreshInterval);
+            $user = $this->getRequest()->getUser();
+
+            if ($user === null) {
+                $speed = 1;
+            } else {
+                $speed = (float) $user->getPreferences()->getValue('icingaweb', 'auto_refresh_speed', 1.0);
+
+                if ($speed <= 0) {
```

Comment:
Don't validate the factor, but the result after applying it. We want a minimum speed but defined in absolute seconds: 5

Though, only if it's the result after the user's speed has been applied. If a controller sets an interval below 5 seconds, it **must** be respected.

Reply:
What for? And why 5?

Reply:
> What for?

Because it's unreasonable to refresh a page at a 1 second interval that's the result of a factor defined in some configuration file. We don't want to allow users to DDOS themselves.

> And why 5?

Because. Well, because there has to be a lower limit, and 5 seconds was the suggestion of a user in the issue and I think it's a sufficiently low/high limit.